src/pykiso/test_coordinator/test_execution.py

```diff
@@ -274,20 +274,70 @@ def parse_test_selection_pattern(pattern: str) -> TestFilterPattern:
     return TestFilterPattern(*parsed_patterns)
 
 
+def find_folders_between_paths(start_path: Path, end_path: Path) -> List[Path]:
+    """Find and return a list of folders between two specified paths.
+
+    :param start_path: the starting path
+    :param end_path: the ending path
+    :return: a list of folders between `start_path` (exclusive) and `end_path` (inclusive). If `end_path` is not
+        a subpath of `start_path`, an empty list is returned.
+    """
+    start_path = start_path
+    end_path = end_path
+    folders_between = []
+
+    while start_path != end_path:
+        folders_between.append(end_path.parent)
+        end_path = end_path.parent
+        if len(end_path.parents) == 1:
+            return []
+    return folders_between
+
+
 def _is_valid_module(start_dir: str, pattern: str) -> bool:
-    """Checks if a given pattern matches invalid python modules in the given directory
+    """Check if the test files found in the specified directory and its subdirectories
+    conform to the requirements of a valid module.
 
     :param start_dir: the directory to search
     :param pattern: pattern that matches the file names
-    :raises InvalidTestModuleName: if a test file name contains a character other
-        than letters, numbers, and _ or starts with a number
     """
-    path = Path(start_dir)
-    file_paths = list(path.glob(pattern))
+    start_dir = Path(start_dir)
+    test_files_found = list(start_dir.glob(f"**/{pattern}"))
+
+    # remove folders
+    test_files_found = [
+        test_file for test_file in test_files_found if test_file.is_file()
+    ]
+
+    # If found test files are in given test suite directory don't check for __init__.py
+    if any(test_file.parent == start_dir for test_file in test_files_found):
+        return True
+
+    # No file matches the test filter pattern
+    if not test_files_found:
+        log.critical(
+            f"Test filter {pattern=} doesn't match any files in folder {start_dir}"
+        )
+        return False
 
-    return bool(
-        file_paths and all([VALID_MODULE_NAME.match(file.name) for file in file_paths])
-    )
+    # Check that all sub folders of a nested testsuite have an __init__.py file
+    testsuite_folders = {filepath.parent for filepath in test_files_found}
```

Comment:
Wouldn't it be easier to directly gather all parent folders of the different files within a set?
`all_parent_folders = set(parent for file in test_files_found for parent in for file.parents)`

Reply:
I need those folders between.. i have seen users who nested testsuite folders and then every folder needs to be a module